Python Simulator/eventgame.py

```diff
@@ -20,30 +21,95 @@ def __init__(self, players=None, num_players=2):
         self.curr_player_num = len(players)  # the player number who's turn it is
         self.final_round_cnt = -1
         self.game_finished = False
-
-    def start_game(self, piece_num, permutation):
+        self.good_moves = []
+
+    def build_board(self, data):
+        board = Board()
+        for piece in data:
+            if not board.add_piece(player_num=piece["player_num"],
+                        piece_num=piece["piece_num"], permutation_index=piece["permutation"],
+                        coordinates=(piece["x"], piece["y"]), check=False):
+                return (board, 0)
+            self.players[piece["player_num"]-1].remove_piece(piece["piece_num"])
+        return (board, 1)
+
+    def execute_moves(self, moves, add_first_move):
+        results = [2] if add_first_move else []
+        for move in moves:
+            if move["type"]==0:
```

Comment:
By convention, operators in Python always have spaces around them. So `if move["type"] == 0:`

Similarly in other locations below.